CoroutineTemplate/app/src/test/java/co/nimblehq/coroutine/test/ViewModelExt.kt

```diff
@@ -0,0 +1,21 @@
+package co.nimblehq.coroutine.test
+
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModel
+import co.nimblehq.coroutine.extension.OverridableLazy
+import kotlin.reflect.KProperty1
+import kotlin.reflect.full.memberProperties
+import kotlin.reflect.jvm.isAccessible
+
+fun <VM : ViewModel, T> T.replace(
+    viewModelDelegate: KProperty1<T, VM>,
+    viewModel: VM
+) {
+    viewModelDelegate.isAccessible = true
+    (viewModelDelegate.getDelegate(this) as OverridableLazy<VM>).implementation = lazy { viewModel }
+}
+
+inline fun <reified T : Fragment, VM> T.getPrivateProperty(name: String): KProperty1<T, VM> =
+    T::class
+        .memberProperties
+        .first { it.name == name } as KProperty1<T, VM>
```

Comment:
Should we use .firstOrNull() instead? Or it definitely found the match name?

Reply:
@lydiasama this is testing, a test throws an exception means it fails and needs to be updated, so we don't need to focus too much on null checking here. Definitely we will execute to call it with correct field name

Reply:
@lydiasama we need to know if the field name is correct or not, so throwing the exception in the test would be helpful 🙏

Reply:
I agree. 👍🏻